crypto/s2n_certificate.h

```diff
@@ -35,6 +35,7 @@ struct s2n_cert {
     s2n_pkey_type pkey_type;
     uint16_t ec_curve_nid;
     s2n_cert_public_key public_key;
+    struct s2n_cert_info info;
```

Comment:
Hmm this is a bit odd. How much information will exist in both s2n_cert_info and s2n_cert? Is ec_curve_nid equivalent to signature_nid?

Reply:
No. `ec_curve_nid` contains the curve information (e.g. p384) but signature information does not (e.g. ECDSA_WITH_SHA256).

This wasn't my first approach, but it was the approach I ended up adopting to try and keep the validator and local cert preference validation mostly similar.